terraform_files/nutanix-ci-machine/main.tf

```
@@ -35,6 +76,7 @@ resource "nutanix_virtual_machine" "vm" {
   num_vcpus_per_socket = var.cores_per_socket
   memory_size_mib      = var.memory
   num_sockets          = var.vcpu
+  guest_customization_cloud_init_user_data = "${data.local_file.cloud_config_file.content_base64}"
```

```suggestion
  locals {
   user_data = <<EOF
#cloud-config
...
EOF
   }
  guest_customization_cloud_init_user_data = base64encode(local.user_data)
```

wouldn't it be simpler?

---

it  is, and  will work - but, I think we should move the cloud config creation to seperated module so we can imported it to others. 

so we will have one config setup for all CI deployment. 

maybe we can play with this config on the next step